### PR TITLE
[WIP] Forum/tree in slate triggered by view

### PR DESCRIPTION
**Pull request by legionth**

### CODE REVIEW

**Comment by legionth** [comment]
Currently WIP. Will refactoring some stuff.

File: src/UI/Implementation/Component/Tree/Node/Factory.php [hunk 1]
```diff
@@ -16,4 +18,8 @@ public function simple(string $label, IIcon $icon=null): INode\Simple
 		return new Simple($label, $icon);
 	}
 
+    public function byline(string $label, string $byline, IIcon $icon = null) : IByline
```
Comment (mjansenDatabay) [inline src/UI/Implementation/Component/Tree/Node/Factory.php]: I suggest `bylined`.

File: src/UI/Implementation/Component/Tree/Node/Factory.php [hunk 2]
```diff
@@ -16,4 +18,8 @@ public function simple(string $label, IIcon $icon=null): INode\Simple
...
+    {
+        return new Byline($label, $byline, $icon);
```
Comment (mjansenDatabay) [inline src/UI/Implementation/Component/Tree/Node/Factory.php]: I suggest `Bylined` as class name.

**Comment by mjansenDatabay** [comment]
LGTM in general! Please create **separate** PRs against the ILIAS `trunk` branch for all changes in `./src/UI`. These should container:

- The introduction o the `Bylined` node type (only interfaces and description/rules) 
- A propsal for an interface change regarding the `Simple` node type, which should be enhanced by an **optional** URL used for a linked type `<a href="[URL]>[Title]</a>`

These PRs should be assigned to Richard and Timon and marked with the `Kitchen Sink` label.

**Pull request closed without merging**